modules/service/src/main/scala/lucuma/odb/graphql/snippet/ObservationSnippet.scala

```diff
@@ -218,31 +270,59 @@ object ObservationSnippet {
     val elaborator = Map[TypeRef, PartialFunction[Select, Result[Query]]](
       QueryType -> {
         case Select("observation", List(
-          ObservationIdBinding("observationId", rOid),
-          BooleanBinding("includeDeleted", rIncludeDeleted)
+          ObservationIdBinding("observationId", rOid)
         ), child) =>
-          (rOid, rIncludeDeleted).parMapN { (oid, includeDeleted) =>
+          rOid.map { oid =>
             Select("observation", Nil,
               Unique(
                 Filter(
                   And(
                     Predicates.hasObservationId(oid),
-                    Predicates.includeDeleted(includeDeleted)
-//                    Predicates.isVisibleTo(user)   // TBD
+                    Predicates.isVisibleTo(user)
                   ),
                   child
                 )
               )
             )
           }
+
+        case Select("observations", List(
+          WhereObservation.Binding.Option("WHERE", rWHERE),
+          ObservationIdBinding.Option("OFFSET", rOFFSET),
+          NonNegIntBinding.Option("LIMIT", rLIMIT),
+          BooleanBinding("includeDeleted", rIncludeDeleted)
+        ), child) =>
+          (rWHERE, rOFFSET, rLIMIT, rIncludeDeleted).parMapN { (WHERE, OFFSET, _, includeDeleted) =>
+            Select("observations", Nil,
+              Filter(
+                And.all(
+                  OFFSET.map(oid => GtEql(UniquePath(List("id")), Const(oid))).getOrElse(True),
+                  Predicates.includeDeleted(includeDeleted),
+                  Predicates.isVisibleTo(user),
+                  WHERE.getOrElse(True)
+                ),
+                child
+                // Limit(
+                //    LIMIT.foldLeft(1000)(_ min _.value),
+                //   child
+                // )
```

Comment:
Pending a Grackle bug fix, I believe.

Reply:
You can fix this by swapping the order; i.e., `Limit(..., Filter(..., child))`